src/System Application/App/AI/src/Azure OpenAI/AOAIAuthorization.Codeunit.al

```diff
@@ -57,6 +62,22 @@ codeunit 7767 "AOAI Authorization"
         Deployment := NewDeployment;
         ApiKey := NewApiKey;
         ManagedResourceDeployment := NewManagedResourceDeployment;
+        MicrosoftManagedAuthorization := true;
+    end;
+
+    [NonDebuggable]
+    procedure SetMicrosoftManagedAuthorization(AOAIAccountName: Text; NewApiKey: SecretText; NewManagedResourceDeployment: Text)
```

Comment:
Mmmm I suggest we keep the pattern that this file was using before (=> the `Set...` functions are just setting values, and then we check that the values are valid inside `IsConfigured`).

This saves us from possible bugs if some other part of the code is changing the authorization variables (and also delays checking the access and the validity of the values until they are actually needed).

Which means, here you just need to set the values for RecourceUtilization, ManagedResourceDeployment, and AOAIAccountName to some global variable. And then you can move the call to `VerifyAOAIAccount` to inside `IsConfigured` instead, and you don't need the three boolean variables.